pyrobosim/pyrobosim/core/world.py

```diff
@@ -1762,7 +1773,20 @@ def check_occupancy(self, pose: Pose | Sequence[float]) -> bool:
         else:
             x, y = pose
 
-        return not bool(shapely.intersects_xy(self.total_internal_polygon, x, y))
+        # Robot assumes all hallway opens
```

Comment:
It seems it might be more efficient to just update the `total_internal_polygon` whenever a robot detects a change in a hallway.

Or in the case of the partial observability being disabled, whenever a hallway state is toggled (which should happen already)

Reply:
Right, so 2 things:

1. I am suggesting that `total_internal_polygon` indeed be changed to per robot, as it would add benefits even beyond this feature.
2. In the partial observability case, it seems like you could maybe also maintain a `total_internal_polygon` variant with only the known closed hallways instead of the ground truth? That would help offset the computation time of performing this check with all the individual polygons.

This also raises a question on whether there's a need for some variation for "check occupancy with ground truth" vs. "check occupancy with known partial state".